src/main/java/com/google/devtools/build/lib/skylarkbuildapi/platform/ConstraintCollectionApi.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.skylarkbuildapi.platform;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.skylarkinterface.Param;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkCallable;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModule;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkModuleCategory;
+import com.google.devtools.build.lib.skylarkinterface.SkylarkValue;
+import com.google.devtools.build.lib.syntax.SkylarkIndexable;
+import javax.annotation.Nullable;
+
+/** Info object representing data about a specific platform. */
+@SkylarkModule(
+    name = "ConstraintCollection",
+    doc = "Provides access to data about a collection of ConstraintValueInfo providers. "
+        + PlatformInfoApi.EXPERIMENTAL_WARNING,
+    category = SkylarkModuleCategory.PROVIDER)
+public interface ConstraintCollectionApi<
+        ConstraintSettingInfoT extends ConstraintSettingInfoApi,
+        ConstraintValueInfoT extends ConstraintValueInfoApi>
+    extends SkylarkIndexable, SkylarkValue {
+
+  @SkylarkCallable(
+      name = "constraint_settings",
+      doc = "The ConstraintSettingInfo values that this collection directly references.",
+      structField = true)
+  ImmutableSet<ConstraintSettingInfoT> constraintSettings();
+
+  @SkylarkCallable(
+      name = "get",
```

Comment:
Can we have a test verifying the skylark API? That, from skylark, one can call .get()? It seems all the current tests only verify that one can call .get() from native code.
(Same goes for constraint_settings)

Reply:
Done.

Reply:
Hmm, I still don't see exactly what I was hoping for -- where are the in-skylark calls to either get() or constraint_settings() ? The new test still doesn't seem to verify either of these methods are callable in starlark...
(Apologies if I'm missing something)

Reply:
Done! This revealed that the return type for constraint_settings was bad, so I am grateful you pushed me to test it. Thanks!

Reply:
Great to hear. Ideally we'd catch this statically or in an automated test, but I suppose we don't yet.
https://github.com/bazelbuild/bazel/issues/6102 to track this.